Tracks: Include blog ID in events when media uploads fail

SiobhyB opened the pull request:
##  Description

When the following track events are fired, they don't currently include a blog's ID, which limits the data we're able to gather around failed media uploads:

* `jpios_media_service_upload_canceled`
* `jpios_media_service_upload_failed`

With this PR, the events are updated so that they'll include a blog's ID when fired.

It's worth highlighting that the blog ID is already collected when `jpios_media_service_upload_started` and `jpios_media_service_upload_successful` fire. So, this PR will only be bringing the cancelled/fired events inline with the started/successful events.

## Testing

**Verify events don't currently include details of blog ID**

* Navigate to the **Live View** tab in Tracks.
* Enter `jpios_media_service_upload_canceled` or `jpios_media_service_upload_failed` into the **Event Name** field, and search for recent events.
* Expand a recent event for further details.
* Verify that blog ID shows up as **null**.

**Verify events will include blog ID with changes from this PR**

* Install the build available in this PR's comments or checkout this branch locally.
* Navigate to the post editor and upload an image from your device's library.
* Before the image finished uploading, tap on it to cancel the process.
* Next, go through the steps to upload another image.
* Before the process finishes, turn off the WiFi so that the upload fails.
* Navigate to the **Live View** tab in Tracks.
* Enter the username associated with your testing account into the **Username, ID, or Email** field. 
* Wait until the `jpios_media_service_upload_canceled` and `jpios_media_service_upload_failed` events show up in your history. _Note, this can take up to 20 minutes._
* When the events show up, verify that the blog ID is visible when expanding their details.

## Regression Notes

1. Potential unintended areas of impact

Any other changes to the way we collect data, beyond adding blog ID to the `*_media_service_upload_canceled` and `*_media_service_upload_failed` events, would be unintentional. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I compared the props for the `*_media_service_upload_canceled` and `*_media_service_upload_failed` events before this change and then after to verify there was nothing unexpected. 

3. What automated tests I added (or what prevented me from doing so)

There are no existing tests in place for events, so I did not add any. As this PR only attaches a new piece of data to existing events, I did not feel a need to further investigate ways to automate testing for events in general.

<hr />

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

SiobhyB committed:
Include blog ID in analytics when uploads fail

SiobhyB committed:
Add optional blog ID arg to WPAnalyticsStat

kean commented on WordPress/Classes/Utility/Analytics/WPAppAnalytics.m:
```diff
@@ -338,14 +338,18 @@ + (void)track:(WPAnalyticsStat)stat withProperties:(NSDictionary *)properties {
                                  @"error_description": err.description
     };
-    [self track:stat withProperties: properties];
+    [self track:stat withProperties: properties withBlogID:blog.dotComID];
```
(Nit) I'd suggest passing only the required parameter – `dotComID` – in the `track` method. It's safer because managed objects have to be thread-confined, so it's generally best to avoid using them if possible.

SiobhyB replied:
Thank you for the note and explanation Alex! I've addressed this in https://github.com/wordpress-mobile/WordPress-iOS/pull/22328/commits/979ee185118ed85e5b7418e6428f91a038aad3cc 🙇‍♀️

kean reviewed (approved):
LGTM, but I left one small comment.

SiobhyB committed:
Refactor func to only pass required dotComID param

This follows the feedback received here: https://github.com/wordpress-mobile/WordPress-iOS/pull/22328/files#r1442167786
